web/proxy.js

```diff
@@ -1,21 +1,18 @@
 const http = require('http');
 const https = require('https');
+const proxyConfig = require('../config/proxyConfig');
 require('dotenv').config();
 
 if (process.env.USE_WEB_PROXY === 'false') {
     process.stdout.write('Skipping proxy as USE_WEB_PROXY was set to false.\n');
     process.exit();
 }
-
-let host = 'www.expensify.com';
-
-// If we are testing against the staging API then we must use the correct host here or nothing with work.
-if (/staging/.test(process.env.EXPENSIFY_URL)) {
-    host = 'staging.expensify.com';
-}
+const host = new URL(process.env.EXPENSIFY_URL || 'https://www.expensify.com').hostname;
+const stagingHost = new URL(process.env.STAGING_EXPENSIFY_URL || 'https://staging.expensify.com').hostname;
+const stagingSecureHost = new URL(process.env.STAGING_SECURE_EXPENSIFY_URL || 'https://staging-secure.expensify.com').hostname;
```

Comment:
Here you have no `dev` in the end of the fallback values

Reply:
This proxy server is meant to be used by people that don't have access to a local (`.dev`) API 
It intercepts requests to the localhost webpack server and redirects them to PROD or STAGING

It's possible (I'm not sure) that internal devs use this and I suppose they will pass their local local server as `EXPENSIFY_URL` in their `.env`. Though I don't really think this is used internally seeing this in `.env.example` - `USE_WEB_PROXY=false`